stablehlo/reference/ProcessGrid.h

```diff
@@ -159,6 +162,11 @@ class ProcessGrid {
   /// processes in the process group.
   std::map<std::pair<ProcessGroup, ChannelId>, RendezvousResult> channels_;
 
+  /// Synchronization primitive used to manage concurrent access to
+  /// `channelLocks_` to prevent multiple processes from creating multiple
+  /// mutexes when the map entry is empty.
+  std::mutex channelLock_;
```

Comment:
Nit: the name is fairly ambiguous with `channelLocks_` defined below. How about `channelsMasterLock` or something?